### ISSUE TITLE
[FR] Method for identifying best-performing model

### ISSUE THREAD

**Issue by jwpestrak** [mlflow/mlflow#1735]
Thank you for submitting an issue. Please refer to our [issue policy](https://www.github.com/mlflow/mlflow/blob/master/ISSUE_POLICY.md)
for information on what types of issues we address.
  
Please do not delete this template unless you are sure your issue is outside its scope.

-------
## Guidelines

Feature requests typically go through the following lifecycle:

1. Submit feature request with high-level description on GitHub issues (this is what you're doing now)
2. Discuss feature request with a committer, who may ask for a more detailed design
3. After discussion & agreement on feature request, start implementation

## Describe the proposal
I propose the addition of functionality to identify the best-performing (by user-specified metric).
For example, consider the code used for model training at `examples/sklearn_elasticnet_wine/train.py`; it would be helpful to have use a function to identify the best-performing model - `my_best_model = mlflow.fetch_best_model(metric=rmse, minimum=True)` - so that I can then pass the model to `log_model()` for future use. 

### Motivation
The motivation is that implementation of a simple search based on metric would promote automation of identification of the best-performing model (instead of having the user manually viewing the tracking UI and specifying the model to pass to `log_model()`).

### Proposed Changes
I propose adding a (admittedly, simplistic) best model identification API to correspond with all `log_model` APIs.

**Comment by apurva-koti** [comment]
One issue is that the definition of "best" can vary depending on the use case. i.e. should the best model be the one that produced the lowest training loss or validation loss? Or test loss?
In the example you linked, we could achieve this by keeping track of whatever metric we want to with a variable, and then logging `lr` with `log_model()`. 
ie. (with some loop)
```python
if loss < best_loss:
    best_loss = loss
    best_model = lr
...
log_model(best_model)
```
  
This is not identical to your use case, but it would be good to know if it covers it.

**Comment by jwpestrak** [comment]
@apurva-koti Your concern about how each each could define "best" is relevant and I appreciate it.  Yes, code that keeps track of a user-specified metric and the model that is associated with its optimum (say, by default, for a validation dataset) would be a nice feature IMO.  If this feature existed, then, when desired by the use, it could shortcut the process of querying and sorting metrics, and identifying the "best" model for logging.

**Comment by sueann** [comment]
Would the `search_runs` API work for you? You can specify the ordering via the `order_by` parameter. https://www.mlflow.org/docs/latest/search-syntax.html#python
